### PR TITLE
Add `all.` prefix to retrieve unhealthy targets

### PR DESCRIPTION
**Pull request by fabxc**
@grobie 

As the `consul_test` is based on the stub, adding a test for `all=true` would only test the functionality of that stub (which would have to be implemented first). I hope it's reasonable to trust that `consul/api`'s passing parameter works as expected.

### CODE REVIEW

**Comment by grobie** [comment]
Thanks, looks great! I agree that it's not reasonable to add a test which tests the stub. To ensure glimpse uses the Consul API correctly also across Consul updates, we have an acceptance test suite. Please add a test case there, it should be similar to:

```golang
		{
			query:   fqdn("all", testCase0.srvAddr, srvZone, dnsZone),
			qtype:   dns.TypeSRV,
			net:     "udp",
			answers: []string{
                          fmt.Sprintf("%s.:%d", nodeName, testCase0.port),
                          fmt.Sprintf("%s.:%d", nodeName, testCase0.port+1),
                        },
		},
```

**Comment by fabxc** [comment]
Updated.

The implementation is still working with a bool, obviously. But the interface is adjusted.

File: agent/helper_test.go [hunk 1]
```diff
@@ -33,7 +37,21 @@ type testStore struct {
...
 
-func (s *testStore) getInstances(srv info) (instances, error) {
+func (s *testStore) getHealthyInstances(srv info) (instances, error) {
+	r, ok := s.instances[srv]
+	if !ok {
+		return nil, newError(errNoInstances, "")
+	}
+	var rc instances
+	for _, inst := range r {
+		if !inst.down {
```
Comment (grobie) [inline agent/helper_test.go]: It doesn't feel right to add the `down` field to a domain struct just for test purposes. Also implementing the filter behavior again in a stub object. What do you think about changing the `testStore` to hold lists for both "healthy" as well as "all" instances and returning both plainly without any logic in these function?

Comment (fabxc) [inline agent/helper_test.go]: Yes, absolutely.
I did unhealthy/healthy lists though as the test initialization remains simpler this way. So `testStore`'s `getAllInstances` concatenates those lists as remaining logic.

**Comment by grobie** [comment]
:+1:

**Pull request merged**